pycrdt_websocket/yroom.py

```diff
@@ -120,22 +120,25 @@ def on_message(self, value: Callable[[bytes], Awaitable[bool] | bool] | None):
         self._on_message = value
 
     async def _broadcast_updates(self):
-        if self.ystore is not None and not self.ystore.started.is_set():
-            self._task_group.start_soon(self.ystore.start)
-
-        async with self._update_receive_stream:
-            async for update in self._update_receive_stream:
-                if self._task_group.cancel_scope.cancel_called:
-                    return
-                # broadcast internal ydoc's update to all clients, that includes changes from the
-                # clients and changes from the backend (out-of-band changes)
-                for client in self.clients:
-                    self.log.debug("Sending Y update to client with endpoint: %s", client.path)
-                    message = create_update_message(update)
-                    self._task_group.start_soon(client.send, message)
-                if self.ystore:
-                    self.log.debug("Writing Y update to YStore")
-                    self._task_group.start_soon(self.ystore.write, update)
+        try:
+            if self.ystore is not None and not self.ystore.started.is_set():
+                self._task_group.start_soon(self.ystore.start)
+
+            async with self._update_receive_stream:
+                async for update in self._update_receive_stream:
+                    if self._task_group.cancel_scope.cancel_called:
+                        return
+                    # broadcast internal ydoc's update to all clients, that includes changes from the
+                    # clients and changes from the backend (out-of-band changes)
+                    for client in self.clients:
+                        self.log.debug("Sending Y update to client with endpoint: %s", client.path)
+                        message = create_update_message(update)
+                        self._task_group.start_soon(client.send, message)
+                    if self.ystore:
+                        self.log.debug("Writing Y update to YStore")
+                        self._task_group.start_soon(self.ystore.write, update)
```

Comment:
Should we stop writing to YStore if it fails?
```suggestion
                    if self.ystore:
                        try:
                            self._task_group.start_soon(self.ystore.write, update)
                            self.log.debug("Writing Y update to YStore")
                        except Exception as e:
                            self.ystore = None
                            self.log.error("Error writing Y update to YStore", exc_info=e)
```

Reply:
If the Update write is caused by a temporary issue and next write restores, would the missing of one Y update cause any issue for future updates or it could auto recovers? I have been purposely adding exception in ystore update flow but I did not notice any obvious issue in UI.